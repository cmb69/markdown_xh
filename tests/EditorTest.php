<?php

namespace Markdown;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeRequest;
use Plib\View;

class EditorTest extends TestCase
{
    private View $view;

    protected function setUp(): void
    {
        $this->view = new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["markdown"]);
    }

    private function sut(): Editor
    {
        return new Editor("./", "./plugins/markdown/", $this->view);
    }

    public function testInitRendersHjs(): void
    {
        $request = new FakeRequest();
        $response = $this->sut()->init([], false, $request);
        Approvals::verifyHtml($response->hjs());
    }

    public function testInitRenderBjs(): void
    {
        $request = new FakeRequest();
        $response = $this->sut()->init([], false, $request);
        Approvals::verifyHtml($response->bjs());
    }

    public function testReplaceReturnCorrectJS(): void
    {
        $request = new FakeRequest();
        $response = $this->sut()->replace("#the-editor", false, $request);
        $this->assertSame('markdown.initById("#the-editor")', $response);
    }

    public function testIncludesOnlyOnce(): void
    {
        $request = new FakeRequest();
        $sut = $this->sut();
        $sut->include($request);
        $response = $sut->include($request);
        $this->assertNull($response->hjs());
        $this->assertNull($response->bjs());
    }
}
