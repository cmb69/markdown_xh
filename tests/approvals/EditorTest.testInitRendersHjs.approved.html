
<link rel="stylesheet" href="./plugins/markdown/markdown/tiny-mde/tiny-mde.min.css">
<script src="./plugins/markdown/markdown/tiny-mde/tiny-mde.min.js"></script>
